### PR TITLE
Add global-tunnel support for proxy (#28)

### LINKED ISSUE #28: GitHub token validation fails in a proxied environment
#### Versions

|                           | Version |
| ------------------------- | ------- |
| verdaccio                 |   3.12.3      |
| verdaccio-github-oauth-ui |    1.7.4     |

#### Expected behavior
Github Oauth callback uses the proxy, executes successfully and returns a JWT when the environment variable `http_proxy` is set.

#### Observed behaviour
Timeout errors when trying to reach api.github.com.

#### Steps to reproduce
1. Go to '/'
2. Click on 'Login'
3. Complete Oauth Login

#### Additional context
The library for requests `got` does not have proxy support by default but we could use something like global-tunnel to inject support dynamically.

### PR DESCRIPTION
**Pull request by seanson**
Fixes #28 
This enables support for any proxy to be discovered automatically via global-tunnel's autoconfig: https://github.com/np-maintain/global-tunnel#auto-config

### CODE REVIEW

**Comment by n4bb12** [comment]
Thanks for this.

`global-tunnel-np` required node < 11. Could you add this info to package.engines and the README?

**Comment by seanson** [comment]
Added info, as well as a note that this plugin is only compatible with Verdaccio 3.

**Comment by n4bb12** [comment]
@seanson hmm I can't see any README changes. There's only one commit.

**Comment by seanson** [comment]
@n4bb12 Apologies, forgot to push the change.

File: README.md [hunk 1]
```diff
@@ -41,6 +41,11 @@ In case you need CLI support for automation purposes, the plugin is also compati
 $ npm install verdaccio-github-oauth-ui
 ```
 
+#### Compatibility
+
+This plugin is currently only compatible with Verdaccio 3.x.
+This plugin supports Node versions 0.6.x - 0.10.x.
```
Comment (n4bb12) [inline README.md]: I'm confused about this version range. Is this semantically the same as in package.engines?

Comment (seanson) [reply README.md]: Fixed typo to match package.engines

**Comment by n4bb12** [comment]
Thank you 🎺

**Comment by n4bb12** [comment]
https://github.com/n4bb12/verdaccio-github-oauth-ui/releases/tag/1.8.0

**Pull request merged**